### PR TITLE
#14643 Easy change layout for error action

### PR DESCRIPTION
**Pull request by swuppio**
Sometimes when we use ErrorAction, we need use different layout, but now it's little bit difficult to do it easily, so this small new feature help to solve problem

```
    public function actions()
    {
        return [
            'error' => [
                'class' => 'yii\web\ErrorAction',
                'layout' => 'error'
            ],
        ];
    }
```

| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #14643

### CODE REVIEW

File: framework/web/ErrorAction.php [hunk 1]
```diff
@@ -98,6 +105,8 @@ public function init()
      */
     public function run()
     {
+        if ($this->layout) $this->controller->layout = $this->layout;
```
Comment (arogachev) [inline framework/web/ErrorAction.php]: Code style must be PSR-2 compatible:

- [Yii 2 Core Framework Code Style](https://github.com/yiisoft/yii2/blob/master/docs/internals/core-code-style.md)
- [PSR-2 - if, elseif, else](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md#51-if-elseif-else)

File: framework/web/ErrorAction.php [hunk 2]
```diff
@@ -74,6 +74,13 @@ class ErrorAction extends Action
      */
     protected $exception;
 
+    /**
+     * @var string|false the name of the layout to be applied to this error action view
+     * @see [[yii\base\Component::$layout]]
+     * @since 2.0.~
```
Comment (arogachev) [inline framework/web/ErrorAction.php]: It's better to specify exact version.

**Review by arogachev — CHANGES_REQUESTED** [review]

File: framework/web/ErrorAction.php (hunk 2 above)
Comment (swods) [inline framework/web/ErrorAction.php]: when i pull requisting it, i didn't know version

**Review by cebe — CHANGES_REQUESTED** [review]
the behavior of `$layout` property of the action should be:

- if `null`, do not change the value in controller
- if anything else (including `false`) set the value on controller.

File: framework/web/ErrorAction.php [hunk 3]
```diff
@@ -74,6 +74,13 @@ class ErrorAction extends Action
      */
     protected $exception;
 
+    /**
+     * @var string|false the name of the layout to be applied to this error action view
+     * @see [[yii\base\Component::$layout]]
```
Comment (cebe) [inline framework/web/ErrorAction.php]: should refer to controller, not component ;) also brackets are not needed.

```
@see \yii\web\Controller::$layout
```

Comment (swods) [inline framework/web/ErrorAction.php]: ohh miss copy

**Comment by cebe** [comment]
I've applied some changes in formatting and changed the check to compare with `null` explicitly.

**Comment by swods** [comment]
heh u was faster, but i wanted to do it like this - !is_null($this->layout), bad?

**Comment by cebe** [comment]
` === null` matches the style we use throughout the framework.

**Comment by klimov-paul** [comment]
https://github.com/yiisoft/yii2/issues/14643#issuecomment-322167870

**Comment by swods** [comment]
© The removal of the tonsils through the anus - best

**Comment by samdark** [comment]
https://github.com/yiisoft/yii2/commit/61b47014a2f68ccb9ed000265a8c413dc4d41af3

Merged. Thank you!

**Pull request closed without merging**